bridge/bridge/deploy/local.py

```diff
@@ -0,0 +1,283 @@
+
+    def check(self):
+        # Check if currently installing
+        if self.install_handle.is_alive() or self.returns[0] is None:
+            logger.info(f"{self.model_version} installing...")
+            return
+
+        # Check if install complete and failed
+        if self.returns[0]:
+            logger.warning(f"{self.model_version} install failed.")
+            destroy_failed_conda(self.model_uri)
+            self._install()
+            return
+
+        # Check if install complete and success, but server not started
+        if self.serve_handle is None:
+            logger.info(f"Serving {self.model_version}")
+            self._serve()
+            return
+
+        # Check if server process exited (restart)
+        if self.serve_handle.poll() is not None:
+            logger.info(
+                f"Detected stopped model deployment {self.model_version}."
+            )
+            destroy_failed_conda(self.model_uri)
+            self._install()
+            self.serve_handle = None
+
+    def _serve(self):
+        model_version = self.model_version
+        path = f"{model_version.model_name}:{model_version.version_id}"
+        stdout = open(f"{path}.stdout", "w")
+        stderr = open(f"{path}.stderr", "w")
+
+        env = {}
+        if env_path := os.environ.get("PATH"):
+            env["PATH"] = env_path
+        if env_home := os.environ.get("HOME"):
+            env["HOME"] = env_home
+
+        self.serve_handle = Popen(
+            [
+                "mlflow",
+                "models",
+                "serve",
+                "--model-uri",
+                self.model_uri,
+                "--port",
+                str(self.port),
+                "--host",
+                "127.0.0.1",
+            ],
+            stdout=stdout,
+            stderr=stderr,
+            preexec_fn=os.setsid,
+            env=env,
+        )
+
+    def destroy(self):
+        if self.serve_handle is not None:
+            os.killpg(os.getpgid(self.serve_handle.pid), signal.SIGTERM)
+            self.serve_handle.wait()
+            self.serve_handle = None
+
+
+class LocalDeployTarget(DeployTarget):
+    target_name = "local"
+
+    @property
+    def target_id(self) -> str:
+        return socket.gethostname()
+
+    def list_models(self) -> List[Model]:
+        # Run reconciliation loop for deployments
+        mvs = list(self.running_models.keys())
+        for mv in mvs:
+            self.running_models[mv].check()
+
+        result = []
+        for model_name, d in self.routes.items():
+            model = Model(name=model_name, versions={})
+            for stage, port in d.items():
+                for mv, md in self.running_models.items():
+                    if port == md.port:
+                        model.versions[stage] = set([mv])
+                        break
+            result.append(model)
+        logger.info(result)
+        return result
+
+    def teardown(self):
+        # TODO: Shutdown Flask
```

Comment:
Is this TODO something that needs to be done before this can merge?

Reply:
It was actually not necessary.